base/shell/explorer/taskswnd.cpp

```diff
@@ -488,26 +488,34 @@ class CTaskSwitchWnd :
 
     HICON GetWndIcon(HWND hwnd)
     {
-        HICON hIcon = 0;
-
-        SendMessageTimeout(hwnd, WM_GETICON, ICON_SMALL2, 0, SMTO_ABORTIFHUNG, 1000, (PDWORD_PTR) &hIcon);
+        const DWORD dwTimeout = 100; // in milliseconds
+        HICON hIcon = NULL;
+        LRESULT bAlive = SendMessageTimeout(hwnd, WM_GETICON, ICON_SMALL2, 0, SMTO_ABORTIFHUNG,
+                                            dwTimeout, (PDWORD_PTR)&hIcon);
         if (hIcon)
             return hIcon;
 
-        SendMessageTimeout(hwnd, WM_GETICON, ICON_SMALL, 0, SMTO_ABORTIFHUNG, 1000, (PDWORD_PTR) &hIcon);
-        if (hIcon)
-            return hIcon;
+        if (bAlive)
+        {
+            bAlive = SendMessageTimeout(hwnd, WM_GETICON, ICON_SMALL, 0, SMTO_ABORTIFHUNG,
+                                        dwTimeout, (PDWORD_PTR)&hIcon);
+            if (hIcon)
+                return hIcon;
+        }
 
-        SendMessageTimeout(hwnd, WM_GETICON, ICON_BIG, 0, SMTO_ABORTIFHUNG, 1000, (PDWORD_PTR) &hIcon);
-        if (hIcon)
-            return hIcon;
+        if (bAlive)
+        {
+            SendMessageTimeout(hwnd, WM_GETICON, ICON_BIG, 0, SMTO_ABORTIFHUNG,
+                               dwTimeout, (PDWORD_PTR)&hIcon);
+            if (hIcon)
+                return hIcon;
+        }
 
         hIcon = (HICON) GetClassLongPtr(hwnd, GCL_HICONSM);
         if (hIcon)
             return hIcon;
 
         hIcon = (HICON) GetClassLongPtr(hwnd, GCL_HICON);
```

Comment:
(Note: since the whole function has been touched anyways, you can take the opportunity to remove the unnecessary extra space between the `(HICON)` cast and the function name.)

Reply:
OK.